p2p/conn/secret_connection.go

```diff
@@ -230,9 +236,12 @@ func (sc *SecretConnection) SetWriteDeadline(t time.Time) error {
 
 func genEphKeys() (ephPub, ephPriv *[32]byte) {
 	var err error
+	// TODO: Probably not a problem but ask Tony: different from the rust implementation (uses x25519-dalek),
+	// we do not "clamp" the private key scalar:
+	// see: https://github.com/dalek-cryptography/x25519-dalek/blob/34676d336049df2bba763cc076a75e47ae1f170f/src/x25519.rs#L56-L74
```

Comment:
@tarcieri: `box.GenerateKey` does not "clamp" the private key. Does this pose a problem? Or should we just ignore this as we will switch to noise soon anyways?

Reply:
Curve25519 canonical scalars are always unclamped. Clamping is performed immediately prior to scalar multiplication (which avoids your exact worry about canonical scalars being "pre-clamped" or not):

https://github.com/golang/crypto/blob/master/curve25519/curve25519.go#L789